workshop/4-Messaging/4.2-Braze.ipynb

```diff
@@ -0,0 +1,522 @@
+    "\n",
+    "![](images/braze-create-template-3.png)\n",
+    "\n",
+    "Braze will validate the file, and return a status window.  Click the Build Email button to continue.\n",
+    "\n",
+    "![](images/braze-create-template-4.png)\n",
+    "\n",
+    "On the next screen, give the template a name, such as `Braze Connected Content Workshop Template`.\n",
+    "\n",
+    "![](images/braze-create-template-5.png)\n",
+    "\n",
+    "On the bottom right of the screen, click the Save Template button.  \n",
+    "\n",
+    "Make sure to click the Save button often, the editor will not auto-save your changes!\n",
+    "\n",
+    "![](images/braze-create-template-6.png)\n",
+    "\n",
+    "Scroll back up and click the Edit Email Body link\n",
+    "\n",
+    "![](images/braze-create-template-7.png)\n",
+    "\n",
+    "This will take you to the HTML template editor.  \n",
+    "\n",
+    "At the very top, you should see the Connected Content call that calls the Personalize recommendation service.\n",
+    "\n",
+    "![](images/braze-create-template-8.png)\n",
+    "\n",
+    "When you are finished looking at your template, click the Done button at the bottom right of the editor screen.\n",
+    "\n",
+    "![](images/braze-create-template-9.png)\n",
+    "\n",
+    "Before navigating away from the template editor, make sure to click the Save Template button on the bottom right of the template window, or your earlier changes will be lost.\n",
+    "\n",
+    "![](images/braze-create-template-10.png)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Create Braze User Profiles\n",
+    "\n",
+    "Before you can run a campaign from Braze, you will need to create a set of demo user profiles with valid emails in Braze. \n",
+    "\n",
+    "For this workshop, we recommend using a set of test emails that (preferably) map to your own email address or to an email address that you can control.\n",
+    "\n",
+    "The following code will use the Braze Users API to create 5 test users in your Braze account.  These users will all have a dummy user ID that starts with `test_user_rds`.  This will allow us to separate these profiles from other user IDs that you may have in Braze, and will allow us to easily delete the test users later.\n",
+    "\n",
+    "For this code to work, you will need to get:\n",
+    "\n",
+    "1. Your Braze endpoint URL, you can look this up [here](https://www.braze.com/docs/api/basics/#api-definitions).  Once you have your endpoint, set the `braze_endpoint_url` variable to the URL from the documentation.\n",
+    "2. Your Braze API key.  We recommend creating an API key just for this workshop, with the `users.track` and the `users.delete` permission at a minimum.  This key can be deleted after your are done with the workshop.  Change the `braze_api_key` variable below to your API key value.\n",
```

Comment:
Will endpoint URL and API key be added as a CFN template parameter and stored as SSM parameter in the future? Maybe for phase 2 when Braze tracking is added to UI?

Reply:
Yes, these will for sure be added as CFN template params and as a SSM param (for the keys) when I add the tracking code.